### PR TITLE
Integrates world.map_cpu to MC

### PR DESCRIPTION
**Pull request by quardbreak**
## Description of changes

_tick.dm: Replaces some define values in _tick.dm with tgstation numbers.

stoplag(): Replaces sleeps with stoplag() in do_after and do_mob.

map_cpu: Integrates it into tick. Plus there is map cpu indicator in MC panel now, which will... indicate it. Informative.

And colorful numbers.

## Why and what will this PR improve

_tick.dm: I'm not sure if this fine to just update MC/tick defines like that. Most updates which bay got for MC was updates straight from tg, and this is why I think some tg's things compatible with nebula.
But while being skeptical, I think values in this PR might be weird and not performant for our codebase, so I asking @Lohikar to check if values are fine.

stoplag(): Lohikar said what stoplag() is fine and better than sleep() in do_after() and do_mob(). Loaf said, that this doesn't seems fine, so I probably will continue to discuss this with them to find answer what to do with it.

map_cpu: The indicator is useful - it gives info how server loaded while dispatching map information to players. 
Seems tgstation integrated map_cpu into tick, so I think this too should be noted before merging.

## Authorship

tgstation for initial tick define values
Vallat for colored CPU numbers idea

### CODE REVIEW

File: code/__defines/_tick.dm [hunk 1]
```diff
@@ -1,13 +1,27 @@
...
+/// Tick limit while initializing
+#define TICK_LIMIT_MC_INIT_DEFAULT (100 - TICK_BYOND_RESERVE)
+
 
-#define TICK_USAGE world.tick_usage //for general usage
-#define TICK_USAGE_REAL world.tick_usage    //to be used where the result isn't checked
+/// for general usage of tick_usage
+#define TICK_USAGE world.tick_usage
+/// to be used where the result isn't checked
+#define TICK_USAGE_REAL world.tick_usage
```
Comment (MistakeNot4892) [inline code/__defines/_tick.dm]: What is the purpose of having two defines with identical values?

File: code/__defines/_tick.dm [hunk 2]
```diff
@@ -1,13 +1,27 @@
...
 
+/// Returns true if tick_usage is above the limit
 #define TICK_CHECK ( TICK_USAGE > Master.current_ticklimit )
-#define CHECK_TICK if TICK_CHECK stoplag()
+/// runs stoplag if tick_usage is above the limit
+#define CHECK_TICK ( TICK_CHECK ? stoplag() : 0 )
```
Comment (MistakeNot4892) [inline code/__defines/_tick.dm:21]: Why change this to a ternary?

File: code/_helpers/mobs.dm [hunk 3]
```diff
@@ -91,7 +91,7 @@
 	var/starttime = world.time
 	. = 1
 	while (world.time < endtime)
-		sleep(1)
+		stoplag(1)
```
Comment (MistakeNot4892) [inline code/_helpers/mobs.dm:94]: These changes are unrelated to the PR. Why do it in this PR? Why do it at all?

Comment (quardbreak) [reply code/_helpers/mobs.dm:94]: They're documented in PR description. If Lohi decide that this should be split - I will remove this.

File: code/__defines/_tick.dm (hunk 1 above)
Comment (quardbreak) [reply code/__defines/_tick.dm]: I don't know. All modern codebases (tg, nebula, bay) have two identical defines. TICK_USAGE_REAL used less and I found only two places where it currently use. TICK_USAGE using more often tho.

File: code/__defines/_tick.dm (hunk 2 above)
Comment (quardbreak) [reply code/__defines/_tick.dm:21]: This adds support for if() statements.

File: code/_helpers/time.dm [hunk 4]
```diff
@@ -138,10 +138,11 @@ var/global/rollovercheck_last_timeofday = 0
 //as sleeps aren't cheap and sleeping only to wake up and sleep again is wasteful
 #define DELTA_CALC max(((max(world.tick_usage, world.cpu) / 100) * max(Master.sleep_delta-1,1)), 1)
 
+//returns the number of ticks slept
 /proc/stoplag(initial_delay)
 	// If we're initializing, our tick limit might be over 100 (testing config), but stoplag() penalizes procs that go over.
 	// 	Unfortunately, this penalty slows down init a *lot*. So, we disable it during boot and lobby, when relatively few things should be calling this.
-	if (!Master || Master.current_runlevel < 3)
+	if (!Master || !(Master.current_runlevel & RUNLEVELS_DEFAULT))
```
Comment (afterthought2) [inline code/_helpers/time.dm]: Doesn't make sense, it's not a bitflag.

File: code/_helpers/mobs.dm (hunk 3 above)
Comment (afterthought2) [reply code/_helpers/mobs.dm:94]: I don't really get it either; it's one thing to stoplag after tick checks, but just a straight stoplag like this feels expensive and pointless. I don't know enough about the internal scheduling to make fully informed comments, though.

EDIT: from talking to Lohikar on discord, it seems doing this is in fact reasonable, if maybe not very consequential.

File: code/__defines/_tick.dm (hunk 1 above)
Comment (afterthought2) [reply code/__defines/_tick.dm]: If you want to keep them distinct, clarify the comment as to their expected usage. `/// to be used where the result isn't checked` is nonsensical to me: result of what? It returns a number; obviously you're checking it, not just calling it for the sake of calling it.

**Comment by quardbreak** [comment]
Not sure how address afterthought's request, because:
1) These two defines existed before me and I didn't change their behavior.
2) I don't know how to write a better explanation in which situation they can be picked.

**Comment by afterthought2** [comment]
I think just combine them into one, unless someone has a coherent explanation of which one should be used when (in which case modify the comments). The comment just doesn't make sense as written, to the point of where deleting it improves clarity.

**Review by afterthought2 — APPROVED** [review]
